Source/Urho3D/Network/AbstractConnection.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020-2025 the rbfx project.
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT> or the accompanying LICENSE file.
+
+#include "Urho3D/Precompiled.h"
+
+#include "Urho3D/Network/AbstractConnection.h"
+
+namespace Urho3D
+{
+
+void AbstractConnection::SendMessage(
+    NetworkMessageId messageId, ConstByteSpan payload, PacketTypeFlags packetType, ea::string_view debugInfo)
+{
+    if (payload.size() > GetMaxMessageSize())
+    {
+        URHO3D_LOGERROR("{}: Message #{} ({} bytes) is too big to send", ToString(), static_cast<unsigned>(messageId),
+            payload.size());
+        return;
+    }
+
+    SendMessageInternal(messageId, payload.data(), payload.size(), packetType);
+
+    Log::GetLogger().Write(GetMessageLogLevel(messageId), "{}: Message #{} ({} bytes) sent{}{}{}{}", ToString(),
+        static_cast<unsigned>(messageId), payload.size(), (packetType & PacketType::Reliable) ? ", reliable" : "",
+        (packetType & PacketType::Ordered) ? ", ordered" : "", debugInfo.empty() ? "" : ": ", debugInfo);
+}
+
+void AbstractConnection::SendMessage(
+    NetworkMessageId messageId, const VectorBuffer& msg, PacketTypeFlags packetType, ea::string_view debugInfo)
+{
+    SendMessage(messageId, msg.GetBuffer(), packetType, debugInfo);
+}
+
+void AbstractConnection::LogReceivedMessage(NetworkMessageId messageId, ea::string_view debugInfo) const
+{
+    Log::GetLogger().Write(GetMessageLogLevel(messageId), "{}: Message #{} received: {}", ToString(),
+        static_cast<unsigned>(messageId), debugInfo);
+}
+
+LogLevel AbstractConnection::GetMessageLogLevel(NetworkMessageId messageId) const
+{
+    static const ea::unordered_set<NetworkMessageId> debugMessages = {
+        MSG_IDENTITY,
+        MSG_SCENELOADED,
+        MSG_REQUESTPACKAGE,
+
+        MSG_LOADSCENE,
+        MSG_SCENECHECKSUMERROR,
+        MSG_PACKAGEINFO,
+
+        MSG_CONFIGURE,
+        MSG_SYNCHRONIZED,
```

Comment:
We should decouple these details from connections. Anything concerning actual behaviors implemented on top of network layer should be elsewhere. (Not a change request, just a general comment for the future)